app/filtering.js

```diff
@@ -727,25 +726,21 @@ module.exports.isResourceEnabled = (resourceName, url, isPrivate) => {
  * @return a promise that always resolves (called on app shutdon so must always)
  */
 module.exports.clearStorageData = () => {
-  let p = Promise.resolve()
   for (let partition in registeredSessions) {
     let ses = registeredSessions[partition]
-    p = p.then(promisify(ses.clearStorageData.bind(ses)).catch(() => {}))
+    setImmediate(ses.clearStorageData.bind(ses))
   }
-  return p
 }
 
 /**
  * Clears all session cache.
  * @return a promise that always resolves (called on app shutdon so must always)
  */
 module.exports.clearCache = () => {
-  let p = Promise.resolve()
   for (let partition in registeredSessions) {
     let ses = registeredSessions[partition]
-    p = p.then(promisify(ses.clearCache.bind(ses)).catch(() => {}))
+    setImmediate(ses.clearCache.bind(ses))
```

Comment:
Since there are docs on the Muon side, I think it would be helpful to put a link to the docs as a comment; ex:
https://github.com/brave/muon/blob/master/docs/api/session.md#sesclearcachecallback

(and if needed, update the docs)